### DISCUSSION TITLE
Option to read assets on startup

### DISCUSSION THREAD

**Discussion by cyberalien** [nitrojs/nitro#2943] [Q&A]
### Describe the feature

Currently when building in Nuxt, Nitro caches list of assets in `server/chunks/nitro/nitro.mjs` and serves only those assets, ignoring actual contents of `public`.

However, I want to keep serving static assets from previous deployment, which are in `public` directory. It would be awesome if there was an option to read `public` directory on startup instead of serving hardcoded set of assets.

### Additional information

- [ ] Would you be willing to help implement this feature?

**Comment by pi0** [comment]
`.output/{public/server}` directory is the artifact of the build, portable and immutable (including `.output/public`), this is by design.

I understand that you want to make public assets from previous builds to the new one in order to allow them being served for clients currently loading older manifest.

Nuxt in particular implemented a build manifest mechanism to detect new builds and vendors such as Vercel [Skew Protection](https://vercel.com/docs/deployments/skew-protection#skew-protection) in order to answer this common problem.

If you have older build files available, there is another way of adding an **additional** public dir so that nitro picks and merges it during build. It merges them.

```js
{
  publicAssets: [{ baseURL: "/_nuxt", dir: "path/to/previous/build/public" }],
}
```
